src/AstAnalyses.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Souffle - A Datalog Compiler
+ * Copyright (c) 2020, The Souffle Developers. All rights reserved.
+ * Licensed under the Universal Permissive License v 1.0 as shown at:
+ * - https://opensource.org/licenses/UPL
+ * - <souffle root>/licenses/SOUFFLE-UPL.txt
+ */
+
+/************************************************************************
+ *
+ * @file AstAnalyses.h
+ *
+ * Define of AST analyses classes
+ *
+ ***********************************************************************/
+
+#pragma once
+
+#include "AstAnalysis.h"
+#include "AstProgram.h"
+#include "AstRelation.h"
+#include "GraphUtils.h"
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <iostream>
+#include <iterator>
+#include <map>
+#include <set>
+#include <stack>
+#include <utility>
+#include <vector>
+
+#include "AstTranslationUnit.h"
+
+namespace souffle {
+
+class AstClause;
+
+/**
+ * Determine the auxiliary arity for relations
+ */
+class AuxiliaryArity : public AstAnalysis {
```

Comment:
The name would be better as `AuxiliaryArityAnalysis`, to better describe the class.